pynest/nest/tests/test_GIDCollection.py

```diff
@@ -601,42 +601,41 @@ def test_get_JSON(self):
         multi_sd = nest.Create('spike_detector', 10)
 
         # Single node, literal parameter
-        pt.assert_frame_equal(single_sd.get('start', output='json'), 0.0)
+        self.assertEqual(single_sd.get('start', output='json'), 0.0)
 
         # Multiple nodes, literal parameter
-        pt.assert_frame_equal(
+        self.assertEqual(
             multi_sd.get('start', output='json'), len(multi_sd) * [0.0])
 
         # Single node, array parameter
         ref_dict = {'start': 0.0, 'n_events': 0}
-        pt.assert_frame_equal(
+        self.assertEqual(
             single_sd.get(['start', 'n_events'], output='json'), ref_dict)
 
         # Multiple nodes, array parameter
         ref_dict = {'start': len(multi_sd) * [0.0],
                     'n_events': len(multi_sd) * [0]}
-        pt.assert_frame_equal(
+        self.assertEqual(
             multi_sd.get(['start', 'n_events'], output='json'), ref_dict)
 
         # Single node, hierarchical with literal parameter
-        pt.assert_frame_equal(
-            single_sd.get('events', 'times', output='json'), [])
+        self.assertEqual(single_sd.get('events', 'times', output='json'), [])
 
         # Multiple nodes, hierarchical with literal parameter
         ref_list = len(multi_sd) * [[]]
-        pt.assert_frame_equal(
+        self.assertEqual(
             multi_sd.get('events', 'times', output='json'), ref_list)
 
         # Single node, hierarchical with array parameter
         ref_dict = {'senders': [], 'times': []}
-        pt.assert_frame_equal(
+        self.assertEqual(
             single_sd.get('events', ['senders', 'times'], output='json'),
             ref_dict)
 
         # Multiple nodes, hierarchical with array parameter
         ref_dict = {'times': len(multi_sd) * [[]],
                     'senders': len(multi_sd) * [[]]}
-        pt.assert_frame_equal(
+        self.assertEqual(
             multi_sd.get('events', ['senders', 'times'], output='json'),
             ref_dict)
```

Comment:
I stopped here, because this output needs to be discussed.
I expected here:
{'senders': [[], [], [], [], [], [], [], [], [], []],
 'times': [[], [], [], [], [], [], [], [], [], []]}
but it returned:
[{'senders': [], 'times': []},
 {'senders': [], 'times': []},
...
 {'senders': [], 'times': []},
 {'senders': [], 'times': []}]

Which should be the correct output?

Reply:
@babsey I agree with you, I expect what you expect (the dict with list of lists), so this must be a bug. `get` should always return a dict, and not a list. I'll take a look at it now!